src/main/kotlin/org/openredstone/trialore/TrialCommand.kt

```diff
@@ -32,9 +33,35 @@ class TrialCommand(
     private val version: String,
 ) : BaseCommand() {
 
+    private fun isTrialOkay(testificate: UUID): Boolean {
+        val trials = trialORE.database.getTrials(testificate)
+        val now = System.currentTimeMillis()
+
+        val recentTrials = trials.map { trialORE.database.getTrialInfo(it) }
+            .filter {
+                val trialTime = it.start.toLong()
+                trialTime >= now - trialORE.config.periodDays * 24 * 60 * 60L
+            }
+            .sortedByDescending { it.start }
+
+        if (recentTrials.size >= trialORE.config.countPerPeriod) {
+            return false
+        }
+
+        val mostRecent = recentTrials.firstOrNull()
+        if (mostRecent != null) {
```

Comment:
Could use something like `.firstOrNull()?.let { it -> etcetcetc }` to "kotlinify" this. https://kotlinlang.org/docs/scope-functions.html#let